### ISSUE TITLE
Images aren't being lazyloaded if they're outside the viewport on IE11

### ISSUE THREAD

**Issue by parkersweb** [verlok/lazyload#184]
Hi there,

I'm having problems with lazyload. On IE11 images inside the Viewport are displayed on page load, but outside the viewport they never get loaded. If I scroll down the page, then resize the browser window those images currently inside the viewport are loaded correctly.

I'm using version 8.6.0. There's no errors being displayed in the console. Other browers (Firefox & Chrome) work fine. This is a typical image tag not being loaded:

````
<img src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
    class="product-item__image lazyload"
    data-srcset="{{ product.featured_image.src | img_url: '260x'}} 260w, {{ product.featured_image.src | img_url: '380x'}} 380w, {{ product.featured_image.src | img_url: '460x'}} 460w"
    sizes="(min-width: 100em) calc(.25 * (100vw - 680px)), (min-width: 81.25em) calc(.33 * (100vw - 680px)), (min-width: 61.875em) calc(.33 * (80vw - 30px)), (min-width: 46.875em) calc(.33 * (100vw - 30px)), calc(50vw - 10px)"
    alt="{{ product.featured_image.alt | escape }}">
<noscript>
    <img class="product-item__image" src="{{ product.featured_image.src | img_url: 400 }}" alt="{{ product.featured_image.alt }}"/>
</noscript>
````

Any idea what I'm doing wrong? Thanks!

**EDIT** Just to add a bit more information - I've also used the callback_enter event to check that the scroll event is firing updates in lazy load - and they're being triggered correctly by IE11 - but the image is never loaded. Those images that aren't loading also still have the "loading" class?

**Comment by parkersweb** [comment]
Thanks to [this](http://www.andreaverlicchi.eu/lazy-load-responsive-images-srcset/) website I've found the solution to this problem. The answer is to re-fire picturefill on the callback_enter event:

````
var myLazyLoad =  new LazyLoad({
    callback_set: function (img) {
        picturefill({
            elements: [img]
        });
    }
});
````
Just in case it helps anyone else :-)

**Comment by verlok** [comment]
Hi there,

Thank you for reporting che issue and the solution. 

Yes, IE11 needs a polyfill for the `srcset` attribute and the polyfill have problems in some cases. 

But I have a question for you now. Why don’t you take out the polyfill and just use the `src` attribute through `data-src`? 

In the end, IE11 users are windows users with large screens so you probably won’t need to serve responsive images to it. Also considering the small amount of users that are still on IE11.

**Comment by parkersweb** [comment]
That sounds perfect - does that have to be done in the callback_set?

I'm assuming it does so that modern browsers don't trigger downloading the full-size image?

Thanks!

**Comment by verlok** [comment]
Correct. My script then sets the `srcset` and the `src` in this exact order so modern browsers read the `srcset` and ignore the `src`.

**Comment by verlok** [comment]
See code here https://github.com/verlok/lazyload/blob/master/src/lazyload.setSources.js

**Comment by parkersweb** [comment]
Works perfectly - and seems more reliable than firing picturefill - thanks!

**Comment by rgfx** [comment]
@parkersweb @verlok 

I am not sure what you guys mean, are you still using picture? I tried change the larger to just src and the img to just src. I makes a lot of sense just to load the largest image in ie, but its not working for me. 

https://codepen.io/rgfx/pen/yKXoQo

lozad has a good way with dealing with ie, add data-iesrc 
 and it will load that image in ie. https://github.com/ApoorvSaxena/lozad.js/blob/master/dist/lozad.js. 

Picture is really good solution for responsive images I hope it can be used without the polyfill. I think the solution that's mentioned is what am looking for, just done know how to implement it,

Thanks

**Comment by parkersweb** [comment]
@rgfx you can check my solution here: https://trymetail.com - I used `<picture>` for the top banner, just `<img>` for the product images. For the `picture` element you can see the `img` tag inside uses `data-src` to provide the IE fallback without a polyfill - more modern browsers use the `data-srcset` elements instead.

**Comment by rgfx** [comment]
@parkersweb. @verlok 

Cool site, I see how its done. The opacity on the image was messing things up for me too.

I noticed a few issues. 

 `data-was-processed=true` is not being fired in IE, that was my problem.  I wonder if its CodePen or did I initialize something wrong? I noticed it was being fired on your site. What are you doing different?

Also if you look at the network in console the `<img>` is loading and so is the main picturefill image. Is this my setup aswell?

https://codepen.io/rgfx/pen/yKXoQo 

Let me know, thanks
